### ISSUE TITLE
[FR] Add ID filter widget to App

### ISSUE THREAD

**Issue by brimoor** [voxel51/fiftyone#1009]
The new `filepath` filter widget in the App is an awesome way to quickly lookup samples by their filepath.

Can we also expose the `id` field and let the user filter based on that? This would allow lookup of arbitrary sample(s) by ID, and would be useful for the same reasons that `filepath` is useful.

One detail here is that the UI should expose strings, and the backend filtering would need to convert to `ObjectId`, as per the other automatic `id (str) <==> _id (ObjectId)` conversions that we expose in our API.

**Discussion**

I wonder if we should introduce a new `DEFAULT FIELDS` section in the filters sidebar where these fields would live? One reason I say this is that these fields will be present on all datasets, so it might be helpful to separate them from `SCALAR FIELDS`, which are fields that the user has manually added to their dataset and thus, if present, may want to see in a dedicated area where "their data" lives.

Also, this would allow a 1-1 correspondence between the fields in the `SCALAR FIELDS` filter section and those in the `Scalars` graph tab. Currently filepath appears in the former but not latter, because filepath histograms are not interesting.

Also, in expanded mode, I would recommend that we continue to include `id` and `filepath` under the `METADATA` section at the top, rather than how filepath is now in the scalar fields section.

<img width="1045" alt="Screen Shot 2021-05-07 at 2 52 32 PM" src="https://user-images.githubusercontent.com/25985824/117496280-d1f40800-af44-11eb-9330-1e3aca2a5768.png">

**Comment by benjaminpkane** [comment]
Hmmm, to support this, I would also like it to be the case that `id` is printed in dataset/view summary. Working through it, noting that this is primarily a `fiftyone.core.odm` issue. The `id` `StringField` would really be the `_id`  `ObjectIdField`...

**Comment by brimoor** [comment]
Yeah I see what you mean. I'd be onboard with exposing the `_id` field of type `ObjectIdField` as an `id` field of type `StringField` in the user-facing schemas.

I think I'd rather continue to hide the fact that IDs are actually stored as `ObjectId`s from the user. The `Document` class does that in a hard-coded way via the getter:

```
@property
def id(self):
    return self._id
```

One nice thing about this is that `ObjectIdField`s are immutable, and the property implementation enforces that immediately at attribute-access time rather than waiting for DB-write time.

**Comment by brimoor** [comment]
Staring at it to see how it feels...
```
Name:        quickstart-video
Media type:  video
Num samples: 10
Persistent:  False
Tags:        []
Sample fields:
    id:       fiftyone.core.fields.StringField
    filepath: fiftyone.core.fields.StringField
    tags:     fiftyone.core.fields.ListField(fiftyone.core.fields.StringField)
    metadata: fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.metadata.Metadata)
Frame fields:
    id:                      fiftyone.core.fields.StringField
    sample_id:               fiftyone.core.fields.StringField
    frame_number:            fiftyone.core.fields.FrameNumberField
    ground_truth_detections: fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.labels.Detections)
```

**Comment by brimoor** [comment]
Maybe we should introduce an optional `public=True` flag on `ObjectIdField` that is valid only when the field's name starts with `_` and allows us to specify whether we'd like the field to appear in schemas without the leading `_`.

I think I'd set this to true for `Sample._id` and `Frame._id` but not `Frame._sample_id`...

**Comment by brimoor** [comment]
@benjaminpkane I can handle adding `id` to the schema. I ran into related needs when working on frames views anyway, so it'll be natural.

**Comment by benjaminpkane** [comment]
Resolved by #1017
